libs/blocks/tabs/tabs.js

```diff
@@ -64,52 +53,106 @@ function changeTabs(e) {
   if(!rows.length) return;
-  const tabList = createTag('div', {role: 'tablist'});
-  const tabListContainer = createTag('div', {class: 'tabList-container container'});
-  const containerWrapper = createTag('div', {class: 'container'});
-  const tabContentContainer = createTag('div', {class: 'tabContent-container'}, containerWrapper);
-  let btnClass = [...element.classList].includes('quiet') ? 'heading-XS' : 'heading-XS'; // tabList size
-  let singleColRows = 0;
-  rows.forEach((row) => { if (row.childElementCount === 1) singleColRows += 1; });
-  if (singleColRows) {
-    const rowHeadler = rows[0].querySelector('h1, h2, h3, h4, h5, h6');
-    rows[0].classList.add('tab-headline', 'container');
-    tabList.setAttribute('aria-label', rowHeadler.textContent);
+
+  // Tab Content
+  const tabContentContainer = createTag('div', {class: 'container'});
+  const tabContent = createTag('div', {class: 'tabContent'}, tabContentContainer);
+  e.append(tabContent);
+
+  // Tab List
+  const tabList = rows[0];
+  const tabId = `tabs-${initCount}`;
+  e.id = tabId;
+  tabList.classList.add('tabList');
+  tabList.setAttribute('role', 'tablist');
+  const tabListContainer = tabList.querySelector(':scope > div');
+  tabListContainer.classList.add('tabList-container');
+  const tabListItems = rows[0].querySelectorAll(':scope li');
+  if (tabListItems) {
+    let btnClass = [...e.classList].includes('quiet') ? 'heading-XS' : 'heading-XS'; // tabList size
+    tabListItems.forEach((item, i) => {
+      const tabName = getStringKeyName(item.textContent);
+      const tabBtnAttributes = {
+        role: 'tab',
+        class: btnClass,
+        id: `tab-${initCount}-${tabName}`,
+        tabindex: (i > 0) ? '0' : '-1',
+      }
+      const tabBtn = createTag('button', tabBtnAttributes);
+      tabBtn.setAttribute('aria-selected', (i === 0) ? 'true' : 'false');
+      tabBtn.setAttribute('aria-controls', `tab-panel-${initCount}-${tabName}`);
+      tabBtn.innerText = item.textContent;
+      tabListContainer.append(tabBtn);
+
+      const tabContentAttributes = {
+        id: `tab-panel-${initCount}-${tabName}`,
+        role: 'tabpanel',
+        tabindex: '0',
+      }
+      const tabListContent = createTag('div', tabContentAttributes);
+      tabListContent.setAttribute('aria-labelledby', `tab-${initCount}-${tabName}`);
+      if(i > 0) tabListContent.setAttribute('hidden', '');
+      tabContentContainer.append(tabListContent);
+    });
+    tabListItems[0].parentElement.remove();
   }
-  const tabRows = element.querySelectorAll(':scope > div:not([class])');
-  tabRows.forEach((row, i) => {
-    const rowTitle = row.querySelector(':scope > div:nth-child(1)');
-    const tabBtnAttributes = {
-      role: 'tab',
-      class: btnClass,
-      id: `tab-${initCount}-${i}`,
-      tabindex: (i > 0) ? '0' : '-1',
+
+  // Tab Config
+  const config = { 'tab-id': initCount };
+  const configRows = e.querySelectorAll(':scope > div:not([class])');
```

Comment:
Is there a more specific selector that's available?

Reply:
Actually, come to think of it, we don't even need another selector. We can just use the rows and splice the first one. This is updated.